azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -180,66 +248,86 @@ def __ensure_azure_slices_exist(self):
-
-            if not os.path.exists(azure_slice_override):
-                self._create_unit_file(azure_slice_override, _AZURE_SLICE_CONTENTS)
-                self._create_unit_file(extensions_slice_override, _EXTENSIONS_SLICE_CONTENTS)
-                drop_in_parent, _ = os.path.split(agent_drop_in_file)
-                if not os.path.exists(drop_in_parent):
-                    fileutil.mkdir(drop_in_parent, mode=0o755)
-                self._create_unit_file(agent_drop_in_file, _AGENT_DROP_IN_CONTENTS)
-                # reload the systemd configuration, but the new slice will not be used until the agent's service restarts
+                return_azure_slice = True
+            else:
+                files_to_create.append((azure_slice, _AZURE_SLICE_CONTENTS))
+
+            if not os.path.exists(vmextensions_slice):
+                files_to_create.append((vmextensions_slice, _VMEXTENSIONS_SLICE_CONTENTS))
+
+            if fileutil.findstr_in_file(agent_unit_file, "Slice=azure.slice"):
+                self._cleanup_unit_file(agent_drop_in_file_slice)
+            else:
+                if not os.path.exists(agent_drop_in_file_slice):
+                    files_to_create.append((agent_drop_in_file_slice, _AGENT_DROP_IN_FILE_SLICE_CONTENTS))
+
+            if fileutil.findstr_in_file(agent_unit_file, "CPUAccounting=yes"):
+                self._cleanup_unit_file(agent_drop_in_file_cpu_accounting)
+            else:
+                if not os.path.exists(agent_drop_in_file_cpu_accounting):
+                    files_to_create.append((agent_drop_in_file_cpu_accounting, _AGENT_DROP_IN_FILE_CPU_ACCOUNTING_CONTENTS))
+
+            if len(files_to_create) > 0:
                 try:
-                    shellutil.run_command(["systemctl", "daemon-reload"])
+                    for path, contents in files_to_create:
+                        self._create_unit_file(path, contents)
+
+                    # reload the systemd configuration, but the new slice will not be used until the agent's service restarts
+                    try:
+                        logger.info("Executing systemctl daemon-reload...")
+                        shellutil.run_command(["systemctl", "daemon-reload"])
+                    except Exception as exception:
+                        self.__log_cgroup_warning("daemon-reload failed: {0}", ustr(exception))
                 except Exception as exception:
-                    self.__log_cgroup_warning("daemon-reload failed: {0}", ustr(exception))
+                    for unit_file in files_to_create:
+                        self._cleanup_unit_file(unit_file)
 
-            return "system.slice"
+            return _AZURE_SLICE if return_azure_slice else "system.slice"
 
         def _create_unit_file(self, path, contents):
             try:
+                parent, _ = os.path.split(path)
+                if not os.path.exists(parent):
+                    fileutil.mkdir(parent, mode=0o755)
```

Comment:
Why 755? Shouldn't it be 644?

Reply:
Is the mode for directories in that location. 644 would be for files, I think.

Reply:
Hmm interesting. But wouldn't 755 make it executable too? Very interesting, will look it up because if this is the norm them I would need to update my PR too (its using 644)

Reply:
Yup, executable means you can traverse the directory

Reply:
woah interesting! I thought traverse was also under `Read`! things you learn everyday haha :)